dongbin-book/04-Simulation/04-1.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @name 상하좌우
+ * @link 동빈북 4-1
+ * @date 2021-10-07
+ * @author sunmon
+ */
+
+const Utils = require("../../lib/utils");
+
+const generator = function* () {
+  const n = yield; // 콘솔에서 입력받은 한 줄 저장
+  const m = yield; // 콘솔에서 입력받은 한 줄 저장
+  const answer = solution(n, m);
+  console.log(answer);
+};
+
+const solution = (n, m) => {
+  const dir = { R: [0, 1], L: [0, -1], U: [-1, 0], D: [1, 0] };
+
+  return m.split(" ").reduce(
+    (prev, cur) => {
+      return [
+        Utils.math.wrap(prev[0] + dir[cur][0], 1, +n),
+        Utils.math.wrap(prev[1] + dir[cur][1], 1, +n),
+      ];
+    },
```

Comment:
```suggestion
    (prev, cur) => ([
        Utils.math.wrap(prev[0] + dir[cur][0], 1, +n),
        Utils.math.wrap(prev[1] + dir[cur][1], 1, +n),
      ])
```

요렇게 써도 되겠지만, 아마 이건 취향이시겠쬬? ㅎㅎ